force-app/main/default/classes/FSL/WorkOrderImportController.cls

```diff
@@ -0,0 +1,456 @@
+        validateWorkOrderData(workOrderData);
+        List<WorkOrderImportModel.WorkOrderExcel> workOrdersWithPropertyIds =
+            addWorkOrderPropertyIds(workOrderData.WorkOrders);
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (workOrdersWithPropertyIds == null) {
+            throw new AuraHandledException('One or more PropertyIds are missing in Database');
+        }
+        List<WorkOrderImportModel.WorkOrderExcel> workOrdersWithPropertyIdsAndContactIdsExcel =
+            addWorkOrderContactIds(workOrdersWithPropertyIds);
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (workOrdersWithPropertyIdsAndContactIdsExcel == null) {
+            throw new AuraHandledException('One or more Caller Emails or Caller Phones are missing in Database');
+        }
+        List<WorkOrderImportModel.WorkOrderExcel> validatedAndPreparedWorkOrders =
+            validateWorkOrders(workOrdersWithPropertyIdsAndContactIdsExcel);
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (validatedAndPreparedWorkOrders == null) {
+            throw new AuraHandledException('Some errors occured trying to create Work Orders. Check provided data');
+        }
+        List<Id> dbWorkOrderExcelAmountIds = createWorkOrders (
+            validatedAndPreparedWorkOrders,
+            workOrdersWithPropertyIdsAndContactIdsExcel
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (dbWorkOrderExcelAmountIds == null) {
+            throw new AuraHandledException('Problem occured during creating Work Orders');
+        }
+        List<Id> dbWorkOrderLineItemIds = createWorkOrderLineItems (
+            workOrderData.WorkOrderLineItems,
+            dbWorkOrderExcelAmountIds
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (dbWorkOrderLineItemIds == null) {
+            throw new AuraHandledException('Problem occured during creating Work Order Line Items');
+        }
+        List<Id> serviceAppointmentIds = createServiceAppointments (
+            workOrderData.ServiceAppointments,
+            dbWorkOrderLineItemIds
+            );
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return JSON.serialize(errorsToFrontList);
+        }
+        if (serviceAppointmentIds == null) {
+            throw new AuraHandledException('Problem occured during creating Service Appointments');
+        }
+        return null;
+    }
+
+    private static List<WorkOrderImportModel.WorkOrderExcel> addWorkOrderPropertyIds (
+        List<WorkOrderImportModel.WorkOrderExcel> workOrderExcelList
+        ) {
```

Comment:
it's hard to read. In this case i'd rather have it in one line, even if it would exceed agreed char limit